### PR TITLE
 [release-4.15] CORS-3169: Add Ingress LB IPs to Infra CR and set DNS unmanaged when BYO DNS is enabled

### PR DESCRIPTION
**Pull request by sadasu**
Manual backport of https://github.com/openshift/cluster-ingress-operator/pull/1016

- Bring in the latest updates to Infrastructure CR from openshift/api
- Update GCPPlatformStatus with the Ingress LB IPs when ClusterHosted DNS is enabled
- Put DNS in Unmanaged state because an in-cluster DNS solution would be in use in place of the platform default DNS.

Based on enhancement: https://github.com/openshift/enhancements/pull/1468

### CODE REVIEW

**Comment by sadasu** [comment]
@Miciah, @gcs278 Could you PTAL?

File: go.mod [hunk 1]
```diff
@@ -115,34 +120,34 @@ require (
...
 )
 
 // These replace stanzas are necessary to import maistra/istio-operator and
 // github.com/operator-framework/operator-sdk.
 replace (
 	bitbucket.org/ww/goautoneg => github.com/munnerz/goautoneg v0.0.0-20120707110453-a547fc61f48d
-	github.com/openshift/api => github.com/openshift/api v0.0.0-20231113114413-39964e6af314
-	k8s.io/client-go => k8s.io/client-go v0.28.0
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20240126160819-e54a015ffc8c
+	k8s.io/client-go => k8s.io/client-go v0.29.0
```
Comment (gcs278) [inline go.mod:152]: I don't think we need to or should rebase the K8S api changes on a backport. `release-4.15` should be 0.28 packages.

Comment (sadasu) [reply go.mod:152]: Moving k8s.io/client-go's version back to v0.28.0 were resulting in these `verify` failures. https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-ingress-operator/1022/pull-ci-openshift-cluster-ingress-operator-release-4.15-verify/1753142958116311040

Comment (gcs278) [reply go.mod:152]: Did you roll back `k8s.io/api v0.29.0` and `k8s.io/apimachinery v0.29.0` as well? That passed verify for me locally. Sorry I meant all K8S packages.

I recommend to start with a clean `go.mod` from the latest commit in release-4.15 without your changes, then update `github.com/openshift/api => github.com/openshift/api v0.0.0-20240126160819-e54a015ffc8c` only and do `go mod tidy` and `go mod vendor`.

You'll get a much cleaner `go.mod`, I only see one other dependency bump (`golang.org/x/exp`). We should try to only bump with is specifically required for limiting the impact of this backport.

Comment (sadasu) [reply go.mod]: Updated with minimal changes to go.mod.

**Comment by sdodson** [comment]
/label approved
I'm approving this backport while the feature is still behind TechPreviewNoUpgrade featuregate. We will require QE to finish their validation and CI testing to confirm that this new feature meets or exceeds 90% pass rate before moving it to on by default. This is not an assertion that this feature will be on by default in 4.15 GA or any time thereafter, that remains to be decided.

**Comment by gcs278** [comment]
No concern from me.
/lgtm

**Comment by sadasu** [comment]
@Miciah would you be able to add the backport-risk-assessed label?

**Pull request merged**